smolyay/samples.py

```diff
@@ -508,3 +508,620 @@ def _create(self):
+
+    options: {None, dict}, optional
+        if method != None, additional parameters passed to the QMCEngine
+    """
+
+    def __init__(self, domain, number_points, seed):
+        super().__init__()
+        self._domain = None
+        self._number_points = None
+        self._seed = None
+
+        self.domain = domain
+        self.number_points = number_points
+        self.seed = seed
+
+    @property
+    def domain(self):
+        """numpy.ndarray: domain of the point set."""
+        return self._domain
+
+    @domain.setter
+    def domain(self, value):
+        domain = numpy.sort(numpy.array(value, ndmin=2), axis=1)
+        if domain.ndim != 2 or domain.shape[1] != 2:
+            raise TypeError("Domain must have size (num_dimensions, 2)")
+        if any(domain[:, 0] >= domain[:, 1]):
+            raise ValueError("Lower bound must be less than upper bound")
+        if not numpy.array_equal(self._domain, domain):
+            self._domain = domain
+            self._valid_cache = False
+
+    @property
+    def number_points(self):
+        """int: random seed for generating points"""
+        return self._number_points
+
+    @number_points.setter
+    def number_points(self, value):
+        number_points = int(value)
+        if self._number_points != number_points:
+            self._number_points = number_points
+            self._valid_cache = False
+
+    @property
+    def seed(self):
+        """int: random seed for generating points"""
+        return self._seed
+
+    @seed.setter
+    def seed(self, value):
+        if not isinstance(value, numpy.random.Generator):
+            value = int(value)
+        if self._seed != value:
+            self._seed = value
+            self._valid_cache = False
+
+    def _create(self):
+        """Generating the Monte Carlo mulitdimensional grid set
+
+        Using the parameters of the class, generate a grid set using
+        the a Monte Carlo/Quasi Monte Carlo method.
+        """
+        self._points = self._get_random_points()
+
+    @abc.abstractmethod
+    def _get_random_points(self):
+        """Generate a set of random points
+
+        Generates a set of random points with a given Monte Carlo/Quasi Monte
+        Carlo method depending on the class.
+
+        Returns
+        -------
+        numpy.ndarray
+            generated random points
+        """
+        pass
+
+
+class UniformRandomPointSet(RandomPointSet):
+    """Generates a grid using a uniform distribution"""
+
+    def _get_random_points(self):
+        lower_bounds = [bound[0] for bound in self.domain]
+        upper_bounds = [bound[1] for bound in self.domain]
+        num_dimensions = len(lower_bounds)
+        p_gen = numpy.random.default_rng(seed=self.seed).uniform(
+            size=(self.number_points, num_dimensions)
+        )
+        return scipy.stats.qmc.scale(p_gen, lower_bounds, upper_bounds)
+
+
+class QMCRandomPointSet(RandomPointSet):
```

Comment:
I don't think this class should be part of the public API. It is just a shim between `RandomPointSet` and the implementations below.
```suggestion
class _QMCRandomPointSet(RandomPointSet):
```